samples/bluetooth/fast_pair/locator_tag/src/fp_adv.c

```diff
@@ -430,52 +484,56 @@ void app_fp_adv_mode_set(enum app_fp_adv_mode adv_mode)
 	}
 }
 
-void app_fp_adv_rpa_rotation_suspend(bool suspended)
+static void app_fp_adv_request_start(void)
 {
-	fp_adv_rpa_rotation_suspended = suspended;
+	bool has_account_key = bt_fast_pair_has_account_key();
+	bool requested_adv_on = false;
+	enum app_fp_adv_mode requested_mode;
+
+	for (int i = 0; i < ARRAY_SIZE(adv_triggers); i++) {
+		if (adv_triggers[i].trigger != NULL) {
+			requested_adv_on |= adv_triggers[i].enabled;
+		}
+	}
+
+	LOG_INF("Fast Pair: %sabling advertising", requested_adv_on ? "en" : "dis");
+
+	if (requested_adv_on) {
+		requested_mode = has_account_key ?
+				 APP_FP_ADV_MODE_NOT_DISCOVERABLE :
+				 APP_FP_ADV_MODE_DISCOVERABLE;
+	} else {
+		requested_mode = APP_FP_ADV_MODE_OFF;
+	}
+
+	app_fp_adv_mode_set(requested_mode);
 }
 
-static void fp_adv_request_handle(enum app_ui_request request)
+void app_fp_adv_request(const char *trigger, bool enable)
 {
-	/* It is assumed that the callback executes in the cooperative
-	 * thread context as it interacts with the FMDN API.
-	 */
-	__ASSERT_NO_MSG(!k_is_preempt_thread());
-	__ASSERT_NO_MSG(!k_is_in_isr());
+	int idx;
+	int ret = app_fp_adv_registered_trigger_find(trigger);
 
-	if (!app_fp_adv_is_init()) {
+	if (ret < 0) {
+		LOG_ERR("Fast Pair: advertising trigger not registered");
+		__ASSERT_NO_MSG(0);
 		return;
```

Comment:
Maybe we can simplify it to `__ASSERT_NO_MSG(ret);` to avoid branching in release configuration?